### DISCUSSION TITLE
Image hashes

### DISCUSSION THREAD

**Discussion by mizaki** [Metron-Project/metroninfo#60] [Ideas] [answered]
I don't see a place for them in the current spec? 

Off the top of my head:
```xml
<hashes>
  <ahash>[hex?]</hash>
  <phash>[hex?]</phash>
</hashes>
```
Or maybe:
```xml
<hash type="ahash">[hex?]</hash>
```

**Comment by bpepple** [comment]
What would that be used for? I'm not quite sure it would be helpful, since the xml is already matched to a source(s).

**Reply by bpepple** [reply]
@mizaki Were you referring to the cover hash? Or every page?

If it was every page,  some of the developers of comic servers didn't think it was terribly useful since they tend to just stat the file for the relevant information.

And like I mentioned above, the cover hash isn't terribly useful *after* the comic has been identified.

Or is there some other use case you have in mind?

**Reply by majora2007** [reply]
Also curious about this as well, I don't really understand a need for this and it would add a large burden on processing time for the tagger itself.

**Reply by mizaki** [reply]
The cover page only.

The processing time I see as the reason to store it. It would make updating faster. (Probably more useful for lower end machines/NAS's.)

This does, I suppose, depend on if metroninfo is meant only for Metron or as a more general standard (I presume the latter). If, hopefully, cover hashes becomes a more standard way to search various information sources, having to only calculate it once would seem like a benefit.

I can see the argument that it's use may be nominal depending on how metroninfo is expected to be used. E.g. One time tag and then ingest into Kav, Kom, Co, etc.

**Reply by bpepple** [reply]
> The processing time I see as the reason to store it. It would make updating faster. (Probably more useful for lower end machines/NAS's.)
> 
> This does, I suppose, depend on if metroninfo is meant only for Metron or as a more general standard (I presume the latter). If, hopefully, cover hashes becomes a more standard way to search various information sources, having to only calculate it once would seem like a benefit.

I guess I don't see how the hash would be used, since this information is being written to xml *after* the comic has been identified, so why need a hash that would be used to identify the comic. 🤔

Maybe if you could give a use case I can wrap my head around the reasoning.

**Reply by mizaki** [reply]
As I mentioned, updating and other sources:
1. Tag with metrontagger/other
2.1. Update at a later date for added/updated information
2.2. Use another information source

If you want to argue having the ID reduces the usefulness, true. I'm not "you have to have this!" :)

**Reply by bpepple** [reply]
> If you want to argue having the ID reduces the usefulness, true. I'm not "you have to have this!" :)

My thinking is having the source `id` is *much* more useful than having a hash, since it would be a definitive match. I'm not inclined to add the hash to the schema, since I don't see it being used much (if at all). What do you think, @majora2007?

**Reply by majora2007** [reply]
I'm in agreement. I'm not against revisiting later, but I still can't see how the cover hash will be used. I believe both Komga and Kavita will take the first file (Kavita searches for cover in the filename first). Even if a hash was available for the cover, it wouldn't really change much from consumption side. 

I could see it being helpful if comparing if a better cover exists from another application, but it doesn't belong in a metadata standard imo.

**Reply by bpepple** [reply, marked as answer]
Ok, so for now, we won't add the cover hash to the schema,  but we are willing to revisit it if there is a compelling argument to add it in the future.
